File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        
        # map of { tuple:List[str]] }
        dict1 = {}
        for word in strs:
            key = tuple(sorted(word))
            if key in dict1.keys():
                dict1[key].append(word)
            else:
                dict1[key] = [word]
        
        return list(dict1.values())

File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        
        
        vector<vector<string>> result;
        
        unordered_map<string, vector<string>> map;
        
        for(string & str: strs) {
            string key = str;
            
            sort(key.begin(), key.end());
            
            map[key].push_back(str);
        }
        
        for(auto [_, anagrams] : map) {
            result.push_back(anagrams);
        }
        
        return result;
    }
                 
};